alts/src/main/java/io/grpc/alts/internal/TsiHandshakeHandler.java

```diff
@@ -137,13 +146,37 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
   }
 
   @Override
-  public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+  public void userEventTriggered(final ChannelHandlerContext ctx, Object evt) throws Exception {
```

Comment:
Question, which of the following can be assumed if `tsiHandshakeHandler.userEventTriggered()` is called multiple times?
1. The argument `ChannelHandlerContext ctx` can be different instances.
2. The argument `evt` of `ProtocolNegotiationEvent` type can be different instances for a fixed `ctx` argument.

Reply:
The `ctx` argument should be the same every time. If not, it is generally a bug, since this is not a `@ChannelHandler.Sharable` handler.

There should only ever be one `ProtocolNegotiationEvent` for this handler instance, which provides this negotiator the results of the previous negotiator. If it happens more than once, then it would be a bug elsewhere.

Reply:
Yeah, that was my understanding.